mesh/nurbs.hpp

```diff
@@ -30,60 +30,118 @@ namespace mfem
 class GridFunction;
 
 
+/** @brief A vector of knots in one dimension, with B-spline basis functions of
+    a prescribed order.
+
+    @note Order is defined in the sense of "The NURBS book" - 2nd ed - Piegl and
+    Tiller, cf. section 2.2.
+*/
 class KnotVector
 {
 protected:
    static const int MaxOrder;
 
-   Vector knot; // Values of knots
-   int Order, NumOfControlPoints, NumOfElements;
+   /// Stores the values of all knots.
+   Vector knot;
+
+   /// Order of the B-spline basis functions.
+   int Order;
+
+   /// Number of control points.
+   int NumOfControlPoints;
+
+   /// Number of elements, defined by distinct knots.
+   int NumOfElements;
 
 public:
-   /// Create KnotVector
+   /// Create an empty KnotVector.
    KnotVector() { }
+
+   /// Create a KnotVector by reading data from stream @a input.
```

Comment:
Should mention here that the stream is expecting two integers:  order and number of control points.

Reply:
Done.